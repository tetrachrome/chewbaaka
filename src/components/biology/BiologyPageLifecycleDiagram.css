/**
 * BiologyPageLifecycleDiagram.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 16, 2020
 * Updated  : Jul 16, 2020
 */

div.BiologyPageLifecycleDiagramOuterContainer {
  margin: 0 auto;
  width: 640px;
}

div.BiologyPageLifecycleDiagramOuterContainer_DebugColor {
  background-color: peachpuff;
}

p.BiologyPageLifecycleDiagramSubtitle {
  font-size: 24px;
}

p.BiologyPageLifecycleDiagramContentText {
  font-size: 18px;
}
