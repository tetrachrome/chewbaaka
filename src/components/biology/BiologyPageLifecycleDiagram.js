/**
 * BiologyPageLifecycleDiagram.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 16, 2020
 * Updated  : Jul 16, 2020
 */

import React from 'react'

import '../shared/ContentPageSharedStyles.css'

import './BiologyPageLifecycleDiagram.css'

import { getElementStyleClassName } from '../../styling/styling'

import image_lifecycle_diagram from './assets/Lifecycle_Diagram_640x675.png'

export default function BiologyPageLifecycleDiagram() {
  return (
    <div className={getElementStyleClassName("BiologyPageLifecycleDiagramOuterContainer")}>
      <h3 className="ContentPageSubsectionTitle TextCentered">The Cycle of Life</h3>
      <p className="BiologyPageLifecycleDiagramSubtitle TextCentered">There are four stages in the lifecycle of the cheetah.</p>
      <p className="BiologyPageLifecycleDiagramContentText TextCentered">
        <b>Stage 1</b>: 90 to 95 days pregnancy; <b>Stage 2</b>: 6 weeks to 18 months;
      </p>
      <p className="BiologyPageLifecycleDiagramContentText TextCentered">
        <b>Stage 3</b>: 18 to 22 months; <b>Stage 4</b>: adult life.
      </p>
      <div className="Centered">
        <img src={image_lifecycle_diagram} alt="lifecycle diagram" />
      </div>
    </div>
  )
}
