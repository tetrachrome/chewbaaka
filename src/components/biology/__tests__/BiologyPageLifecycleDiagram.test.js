/**
 * BiologyPageLifecycleDiagram.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 16, 2020
 * Updated  : Jul 16, 2020
 */

import React from 'react'

import { render } from '@testing-library/react'

import renderer from 'react-test-renderer'

import BiologyPageLifecycleDiagram from '../BiologyPageLifecycleDiagram'

test('renders BiologyPageLifecycleDiagram component', () => {
  render(
    <BiologyPageLifecycleDiagram />
  );
});

test('BiologyPageLifecycleDiagram component snapshot', () => {
  const tree = renderer
    .create(
      <BiologyPageLifecycleDiagram />
    ) .toJSON();
  expect(tree).toMatchSnapshot();
});
