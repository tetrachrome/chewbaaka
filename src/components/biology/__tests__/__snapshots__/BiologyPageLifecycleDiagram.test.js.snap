// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BiologyPageLifecycleDiagram component snapshot 1`] = `
<div
  className="BiologyPageLifecycleDiagramOuterContainer"
>
  <h3
    className="ContentPageSubsectionTitle TextCentered"
  >
    The Cycle of Life
  </h3>
  <p
    className="BiologyPageLifecycleDiagramSubtitle TextCentered"
  >
    There are four stages in the lifecycle of the cheetah.
  </p>
  <p
    className="BiologyPageLifecycleDiagramContentText TextCentered"
  >
    <b>
      Stage 1
    </b>
    : 90 to 95 days pregnancy; 
    <b>
      Stage 2
    </b>
    : 6 weeks to 18 months;
  </p>
  <p
    className="BiologyPageLifecycleDiagramContentText TextCentered"
  >
    <b>
      Stage 3
    </b>
    : 18 to 22 months; 
    <b>
      Stage 4
    </b>
    : adult life.
  </p>
  <div
    className="Centered"
  >
    <img
      alt="lifecycle diagram"
      src="Lifecycle_Diagram_640x675.png"
    />
  </div>
</div>
`;
